watchtower/lookout/justice_descriptor.go

```diff
@@ -104,20 +105,35 @@ func (p *JusticeDescriptor) commitToRemoteInput() (*breachedInput, error) {
 		return nil, err
 	}
 
-	// Since the to-remote witness script should just be a regular p2wkh
-	// output, we'll parse it to retrieve the public key.
-	toRemotePubKey, err := btcec.ParsePubKey(toRemoteScript, btcec.S256())
-	if err != nil {
-		return nil, err
-	}
-
-	// Compute the witness script hash from the to-remote pubkey, which will
-	// be used to locate the input on the breach commitment transaction.
-	toRemoteScriptHash, err := input.CommitScriptUnencumbered(
-		toRemotePubKey,
+	var (
+		toRemoteScriptHash []byte
```

Comment:
FWIW, when in the non nested-p2wsh land I usually like to refer to this as the "witness script" to distinguish things a bit.

Reply:
hmm, technically this is referred to as the witness _program_ in BIP 143. IMO the term witness script is ambiguous because we use it interchangeably as as you said for witness program, but also for redeem script.

Reply:
Hmm, from my PoV (non-blocking here):
  * witness program -> the pkScript iself 
     * this _is_ also used as the _redeem script_ for nested p2wsh 
  * witness script -> the top element in the witness stack for p2wsh sends
      * this is typically what's signed, but there's a quirk where Script "injects" an "unrolled p2wkh" script